backend/ops_api/ops/resources/agreements.py

```diff
@@ -247,22 +234,38 @@ def patch(self, id: int) -> Response:
 class AgreementListAPI(BaseListAPI):
     def __init__(self, model: BaseModel = Agreement):
         super().__init__(model)
-        self._post_schema_contract = desert.schema(ContractAgreementRequestBody)
-        self._post_schema_grant = desert.schema(GrantAgreementRequestBody)
-        self._get_schema = desert.schema(QueryParameters)
+        self._schema_contract = desert.schema(ContractAgreementData)
+        self._schema_grant = desert.schema(GrantAgreementData)
 
     @staticmethod
-    def _get_query(search=None, research_project_id=None):
-        stmt = select(Agreement).order_by(Agreement.id)
+    def _get_query(args):
+        polymorphic_agreement = with_polymorphic(Agreement, [ContractAgreement, GrantAgreement])
+        stmt = select(polymorphic_agreement).order_by(Agreement.id)
         query_helper = QueryHelper(stmt)
 
-        if search is not None and len(search) == 0:
-            query_helper.return_none()
-        elif search:
-            query_helper.add_search(Agreement.name, search)
-
-        if research_project_id:
-            query_helper.add_column_equals(Agreement.research_project_id, research_project_id)
+        match args:
+            case {"search": search, **filter_args} if not search:
```

Comment:
Yeah, because you can filter only based on exact values, while search allows for only partial values. Useful for the really-long-names they might use.